capplets/time-admin/src/time-tool.c

```diff
@@ -0,0 +1,309 @@
+/*  time-admin 
+*   Copyright (C) 2018  zhuyaliang https://github.com/zhuyaliang/
+*
+*   This program is free software: you can redistribute it and/or modify
+*   it under the terms of the GNU General Public License as published by
+*   the Free Software Foundation, either version 3 of the License, or
+*   (at your option) any later version.
+
+*   This program is distributed in the hope that it will be useful,
+*   but WITHOUT ANY WARRANTY; without even the implied warranty of
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*   GNU General Public License for more details.
+
+*   You should have received a copy of the GNU General Public License
+*   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "time-tool.h"
+
+struct tm *GetCurrentTime(void)
+{
+	time_t tt;
+	tzset();
+	tt=time(NULL);
+	
+    return localtime(&tt);
+}
+static void UpdateDate(TimeAdmin *ta,gboolean state)
+{
+    struct tm *LocalTime;
```

Comment:
**cppcheck warning**
Id: variableScope
The scope of the variable 'LocalTime' can be reduced.

Reply:
I don't think it needs to be changed. It's just a matter of programming habits.